src/prefect/agent/agent.py

```diff
@@ -673,7 +673,7 @@ def mark_failed(self, flow_run: GraphQLResult, exc: Exception) -> None:
             version=flow_run.version,
             state=Failed(message=str(exc)),
         )
-        self.logger.error("Error while deploying flow: {}".format(repr(exc)))
+        self.logger.error("Error while deploying flow: {}".format(repr(exc)), exc_info=exc)
```

Comment:
More helpful logs by printing the stack trace:

Old behaviour (Example):
```
ERROR - SharepointDockerAgent | Error while deploying flow: APIError(HTTPError('500 Server Error: Internal Server Error for url: http+docker://localhost/v1.41/containers/create'))
```

New behaviour (example):
```
prefect-agent            | [2021-02-02 09:16:06,419] ERROR - SharepointDockerAgent | Error while deploying flow: APIError(HTTPError('500 Server Error: Internal Server Error for url: http+docker://localhost/v1.41/containers/create'))
prefect-agent            | Traceback (most recent call last):
prefect-agent            |   File "/opt/runtime/python_env/lib/python3.8/site-packages/docker/api/client.py", line 268, in _raise_for_status
prefect-age
```
nt            |     response.raise_for_status()
prefect-agent            |   File "/opt/runtime/python_env/lib/python3.8/site-packages/requests/models.py", line 943, in raise_for_status
prefect-agent            |     raise HTTPError(http_error_msg, response=self)
prefect-agent            | requests.exceptions.HTTPError: 500 Server Error: Internal Server Error for url: http+docker://localhost/v1.41/containers/create
prefect-agent            | 
prefect-agent            | During handling of the above exception, another exception occurred:
prefect-agent            | 
prefect-agent            | Traceback (most recent call last):
prefect-agent            |   File "/opt/runtime/python_env/lib/python3.8/site-packages/prefect/agent/agent.py", line 406, in deploy_and_update_flow_run
prefect-agent            |     deployment_info = self.deploy_flow(flow_run)
prefect-agent            |   File "/opt/runtime/python_env/lib/python3.8/site-packages/prefect/agent/docker/agent.py", line 436, in deploy_flow
prefect-agent            |     container = self.docker_client.create_container(
prefect-agent            |   File "/opt/runtime/python_env/lib/python3.8/site-packages/docker/api/container.py", line 430, in create_container
prefect-agent            |     return self.create_container_from_config(config, name)
prefect-agent            |   File "/opt/runtime/python_env/lib/python3.8/site-packages/docker/api/container.py", line 441, in create_container_from_config
prefect-agent            |     return self._result(res, True)
prefect-agent            |   File "/opt/runtime/python_env/lib/python3.8/site-packages/docker/api/client.py", line 274, in _result
prefect-agent            |     self._raise_for_status(response)
prefect-agent            |   File "/opt/runtime/python_env/lib/python3.8/site-packages/docker/api/client.py", line 270, in _raise_for_status
prefect-agent            |     raise create_api_error_from_http_exception(e)
prefect-agent            |   File "/opt/runtime/python_env/lib/python3.8/site-packages/docker/errors.py", line 31, in create_api_error_from_http_exception
prefect-agent            |     raise cls(e, response=response, explanation=explanation)
prefect-agent            | docker.errors.APIError: 500 Server Error for http+docker://localhost/v1.41/containers/create: Internal Server Error ("invalid volume specification: '/home/peter/prefect_run/download:.:rw': invalid mount config for type "bind": invalid mount path: '.' mount path must be absolute")

```

Reply:
Nice! Much more descriptive